src/index.ts

```diff
@@ -22,10 +22,9 @@ interface Ctx {
 }
 
 function isNewtype(tpe: Tpe): Reader<Ctx, boolean> {
-  return ask<Ctx>().map(({ models }) => {
-    const model = models.find(m => m.name === tpe.name);
-    return !!(model && 'isValueClass' in model && model.isValueClass);
-  });
+  return ask<Ctx>().map(({ models }) =>
+    findFirst(models, m => m.name === tpe.name && 'isValueClass' in m && m.isValueClass).isSome()
```

Comment:
looking at this, probably it can be written using `.some` from the stdlib.

```typescript
models.some(m => m.name === tpe.name && 'isValueClass' in m && m.isValueClass)
```